src/shapes/Group.spec.ts

```diff
@@ -15,6 +16,101 @@ describe('Group', () => {
     expect(group._objects).toHaveLength(3);
   });
 
+  it('fromObject restores values as they are, ignoring specific width/height/top/left that could come from layout', async () => {
+    const objectData = {
+      width: 2,
+      height: 3,
+      left: 6,
+      top: 4,
+      strokeWidth: 0,
+      objects: [
+        new Rect({ width: 100, height: 100, top: 0, left: 0, strokeWidth: 0 }),
+      ],
+    };
+
+    const group = await Group.fromObject(objectData);
+    expect(group.width).toBe(objectData.width);
+    expect(group.height).toBe(objectData.height);
+    expect(group.left).toBe(objectData.left);
+    expect(group.top).toBe(objectData.top);
+    group.triggerLayout();
+    expect(group.width).toBe(100);
+    expect(group.height).toBe(100);
+  });
+
+  describe('With fit-content layout manager', () => {
+    it('Group initialization will calculate correct width/height ignoring passed width and height', async () => {
+      const objectOptions = {
+        width: 2,
+        height: 3,
+        left: 6,
+        top: 4,
+        strokeWidth: 0,
+      };
+
+      const group = new Group(
+        [
+          new Rect({
+            width: 100,
+            height: 100,
+            top: 0,
+            left: 0,
+            strokeWidth: 0,
+          }),
+        ],
+        objectOptions
+      );
+      expect(group.width).toBe(100);
+      expect(group.height).toBe(100);
+    });
+    it('Group initialization will calculate width/height ignoring the passed one', async () => {
+      const objectOptions = {
+        width: 2,
+        height: 3,
+        left: 6,
+        top: 4,
+        strokeWidth: 0,
+      };
+
+      const group = new Group(
+        [
+          new Rect({
+            width: 100,
+            height: 100,
+            top: 0,
+            left: 0,
+            strokeWidth: 0,
+          }),
+        ],
+        objectOptions
+      );
+      expect(group.left).toBe(6);
+      expect(group.top).toBe(4);
```

Comment:
seems off to me

Reply:
this is correct, the layout manager has been built like this:
```js
    if (context.type === LAYOUT_TYPE_INITIALIZATION) {
      // TODO: what about strokeWidth?
      target.set({
        left:
          context.x ?? nextCenter.x + size.x * resolveOrigin(target.originX),
        top: context.y ?? nextCenter.y + size.y * resolveOrigin(target.originY),
      });
    }
```
initialization will take position from context if available,

and in constructor:

```js
    layoutManager.performLayout({
      type: LAYOUT_TYPE_INITIALIZATION,
      target: this,
      targets: [...objects],
      x: options.left,
      y: options.top,
    });
```

we pass top and left as context.